pkg/proxy/ipvs/proxier.go

```diff
@@ -1944,59 +1949,58 @@ func (proxier *Proxier) deleteEndpointConnections(connectionMap []proxy.ServiceE
 	}
 }
 
-func (proxier *Proxier) syncService(svcName string, vs *utilipvs.VirtualServer, bindAddr bool, bindedAddresses sets.String) error {
-	appliedVirtualServer, _ := proxier.ipvs.GetVirtualServer(vs)
-	if appliedVirtualServer == nil || !appliedVirtualServer.Equal(vs) {
-		if appliedVirtualServer == nil {
+func (proxier *Proxier) syncService(svcName string, wantedVS, appliedVS *utilipvs.VirtualServer, bindAddr bool, bindedAddresses sets.String) (*utilipvs.VirtualServer, error) {
+	if appliedVS == nil || !appliedVS.Equal(wantedVS) {
+		if appliedVS == nil {
 			// IPVS service is not found, create a new service
-			klog.V(3).Infof("Adding new service %q %s:%d/%s", svcName, vs.Address, vs.Port, vs.Protocol)
-			if err := proxier.ipvs.AddVirtualServer(vs); err != nil {
+			klog.V(3).Infof("Adding new service %q %s:%d/%s", svcName, wantedVS.Address, wantedVS.Port, wantedVS.Protocol)
+			if err := proxier.ipvs.AddVirtualServer(wantedVS); err != nil {
 				klog.Errorf("Failed to add IPVS service %q: %v", svcName, err)
-				return err
+				return nil, err
 			}
 		} else {
 			// IPVS service was changed, update the existing one
 			// During updates, service VIP will not go down
 			klog.V(3).Infof("IPVS service %s was changed", svcName)
-			if err := proxier.ipvs.UpdateVirtualServer(vs); err != nil {
+			if err := proxier.ipvs.UpdateVirtualServer(wantedVS); err != nil {
 				klog.Errorf("Failed to update IPVS service, err:%v", err)
-				return err
+				return nil, err
 			}
 		}
+		var err error
+		appliedVS, err = proxier.ipvs.GetVirtualServer(wantedVS)
```

Comment:
prefer to name this `updatedVS` and the parameter `appliedVS` to `currentVS`

Reply:
Done.